pkg/providers/tinkerbell/hardware/machine.go

```diff
@@ -110,13 +110,22 @@ const NameserversSeparator = "|"
 // Nameservers is a custom type that can unmarshal a CSV representation of nameservers.
 type Nameservers []string
 
+func (n *Nameservers) ToString() string {
+	return strings.Join(*n, NameserversSeparator)
+}
+
 // UnmarshalCSV unmarshalls s where is is a list of nameservers separated by NameserversSeparator.
 func (n *Nameservers) UnmarshalCSV(s string) error {
 	servers := strings.Split(s, NameserversSeparator)
 	*n = append(*n, servers...)
 	return nil
 }
 
+// MarshalCSV marshalls Nameservers into a string list of nameservers separated by NameserversSeparator.
```

Comment:
I've just added this on a different PR for the CSV parsing as I also needed it in test FYI in-case mine goes in before.